### PR TITLE
[JENKIKNS-54355] Dynamically load java.sql classes

### PR DESCRIPTION
**Pull request by alecharp**
[JENKINS-54355](https://issues.jenkins-ci.org/browse/JENKINS-54355)

Tag.java has references to classes which are not longer available in Java 11, without a module addition. This could cause problem when running this plugin on a Jenkins with Java 11.

@jenkinsci/java11-support

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/shaded/org/yaml/snakeyaml/nodes/Tag.java [hunk 1]
```diff
@@ -61,8 +64,13 @@
...
         timestampSet.add(Date.class);
-        timestampSet.add(java.sql.Date.class);
-        timestampSet.add(Timestamp.class);
+        try {
+            timestampSet.add(ClassLoader.getSystemClassLoader().loadClass("java.sql.Date"));
+            timestampSet.add(ClassLoader.getSystemClassLoader().loadClass("java.sql.Timestamp"));
+        } catch (ClassNotFoundException e) {
+            LOG.log(Level.INFO, "java.sql.Date and / or java.sql.Timestamp are not available. " +
+                  "You are probably on Java 11, if so, it's ok.", e);
+        }
```
Comment (batmat) [inline src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/shaded/org/yaml/snakeyaml/nodes/Tag.java]: Upside: this looks like you got the right fix :-) https://bitbucket.org/asomov/snakeyaml/src/d318aef91b9df07b369eba6c553a8457b05e0726/src/main/java/org/yaml/snakeyaml/nodes/Tag.java?at=default&fileviewer=file-view-default#Tag.java-64:70

Downside: I actually think the right fix would be to upgrade snakeyaml. But seems like this is not yet released. So this looks fine.

I commented in https://bitbucket.org/asomov/snakeyaml/issues/422/provide-a-stripped-down-library-for-just to ask for a release.

I think in the meantime this fix looks correct.

File: Jenkinsfile [hunk 2]
```diff
@@ -1 +1,11 @@
-buildPlugin(findbugs: [run: true, archive:true, unstableTotalAll: "0"])
+buildPlugin(
+    findbugs: [run: true, archive:true, unstableTotalAll: "0"],
+    configurations: [
+        [ platform: "linux", jdk: "8", jenkins: null ],
+        [ platform: "windows", jdk: "8", jenkins: null ],
+        [ platform: "linux", jdk: "8", jenkins: "2.150.1", javaLevel: "8" ],
+        [ platform: "windows", jdk: "8", jenkins: "2.150.1", javaLevel: "8" ],
```
Comment (batmat) [inline Jenkinsfile]: :question: The addition below for JDK11 does make sense. 

But the two lines above adding `jenkins.version=2.150.1` seem unrelated. I mean, why not, but it _looks_ slightly out of scope here.

Comment (alecharp) [reply Jenkinsfile]: anyway, I think this is not working as the build only ran on linux/windows 8, nothing in 11.

File: src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/shaded/org/yaml/snakeyaml/nodes/Tag.java (hunk 1 above)
Comment (alecharp) [reply src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/shaded/org/yaml/snakeyaml/nodes/Tag.java]: Upgrading snakeyaml should be done in a dedicated PR imo, even if it was released with the fix.

Comment (batmat) [reply src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/shaded/org/yaml/snakeyaml/nodes/Tag.java]: Yeah, but then upgrading snakeyaml **would be** the fix. But as it's not released, that's fine.

Comment (batmat) [reply src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/shaded/org/yaml/snakeyaml/nodes/Tag.java]: Also, got feedback from the snakeyaml maintainer in the issue linked above: release with this fix is reflected in February.

**Comment by batmat** [comment]
Given since this was filed, we understood a bit more and didn't actually see this surface in a normal Jenkins environment (i.e. not in test mode, PCT, JenkinsfileRunner, etc.), I'm inclined to now say we should close it (until at least we possibly see this surface actually in normal running conditions).

**Comment by alecharp** [comment]
I have no hard feelings about that.

File: Jenkinsfile (hunk 2 above)
Comment (rsandell) [reply Jenkinsfile]: I've changed the Jenkinsfile after the merge so that it only runs the jdk 11 configurations if anyone of you guys are the PR author.
Too many test failures :)
